guides/index-template.md

```diff
@@ -19,186 +17,169 @@ var config = new ConnectionSettings(node)
-                    .Name("published_on"))
-                .Number(n => n
-                    .Name("pages")
-                    .Type(NumberType.Integer))))));
-
-// Create an index template for "books"
-var createBooksTemplateResponse = client.LowLevel.Indices.PutTemplateV2ForAll<VoidResponse>("books", PostData.Serializable(new
-{
-    index_patterns = new[] { "books-*" },
-    composed_of = new[] { "books_mappings" },
-    priority = 0,
-    template = new
-    {
-        settings = new
-        {
-            index = new
-            {
-                number_of_shards = 3,
-                number_of_replicas = 0
-            }
-        }
-    }
-}));
-
-// Create an index template for "books-fiction"
-var createBooksFictionTemplateResponse = client.LowLevel.Indices.PutTemplateV2ForAll<VoidResponse>("books-fiction", PostData.Serializable(new
-{
-    index_patterns = new[] { "books-fiction-*" },
-    composed_of = new[] { "books_mappings" },
-    priority = 1,
-    template = new
-    {
-        settings = new
-        {
-            index = new
-            {
-                number_of_shards = 1,
-                number_of_replicas = 1
-            }
-        }
-    }
-}));
+var putComponentTemplate = await client.Cluster.PutComponentTemplateAsync("books_mappings", d => d
+	.Template(t => t
+		.Map<Book>(m => m
+			.Properties(p => p
+				.Text(f => f.Name(b => b.Title))
+				.Text(f => f.Name(b => b.Author))
+				.Date(f => f.Name(b => b.PublishedOn))
+				.Number(f => f.Name(b => b.Pages).Type(NumberType.Integer))
+			))));
+Debug.Assert(putComponentTemplate.IsValid, putComponentTemplate.DebugInformation);
+
+putTemplate = await client.Indices.PutComposableTemplateAsync("books", d => d
+	.IndexPatterns("books-*")
+	.Priority(0)
+	.ComposedOf("books_mappings")
+	.Template(t => t
+		.Settings(s => s
+			.NumberOfShards(3)
+			.NumberOfReplicas(0))));
+Debug.Assert(putTemplate.IsValid, putTemplate.DebugInformation);
+
+putTemplate = await client.Indices.PutComposableTemplateAsync("books-fiction", d => d
+	.IndexPatterns("books-fiction-*")
+	.Priority(1) // higher priority than the `books` template
+	.ComposedOf("books_mappings")
+	.Template(t => t
+		.Settings(s => s
+			.NumberOfShards(1)
+			.NumberOfReplicas(1))));
+Debug.Assert(putTemplate.IsValid, putTemplate.DebugInformation);
 ```
 
 When we create an index named `books-fiction-horror`, OpenSearch will apply the `books-fiction-*` template's settings, and `books_mappings` template mappings to the index:
 
 ```csharp
-client.Indices.Create("books-fiction-horror");
-var getResponse = client.Indices.Get("books-fiction-horror");
-Console.WriteLine(getResponse.Indices["books-fiction-horror"].Settings.NumberOfShards); // 1 Console.WriteLine(getResponse.Indices["books-fiction-horror"].Mappings.Properties["pages"].Type); // integer 
+createIndex = await client.Indices.CreateAsync("books-fiction-horror");
+Debug.Assert(createIndex.IsValid, createIndex.DebugInformation);
```

Comment:
I think for a guide we may want to `Console.WriteLine` instead of `Debug.Assert`.